File: solutions/lua/difference-of-squares/1/difference-of-squares.lua
local function square_of_sum(n)
  s=0
  for i=1,n,1
  do 
    s=s+i
  end
  return s*s
end

local function sum_of_squares(n)
  s=0
  for i=1,n,1
  do 
    s=s+i*i
  end
  return s
end

local function difference_of_squares(n)
  return square_of_sum(n) - sum_of_squares(n)
end

return {
  square_of_sum = square_of_sum,
  sum_of_squares = sum_of_squares,
  difference_of_squares = difference_of_squares
}

File: solutions/go/difference-of-squares/1/difference_of_squares.go
// Package diffsquares calculates the difference between the square of the sum of first nth numbers in the series and the sum of the square of the same numbers.
package diffsquares

// SquareOfSum returns the square of the sum of the series of numbers from 1 to its argument n
func SquareOfSum(n int) int {
	var sum int
	for i := 1; i <= n; i++ {
		sum += i
	}
	return sum * sum
}

// SumOfSquares returns the sum of the series of numbers from 1 to its argument n squared
func SumOfSquares(n int) int {
	var sum int
	for i := 1; i <= n; i++ {
		sum += i * i
	}
	return sum
}

// Difference returns the sum of the series of numbers from 1 to its argument n squared
func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}
